include/tvm/support/random_engine.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file random_engine.h
+ * \brief Random number generator, for Sampler and Sampling functions.
+ */
+
+#ifndef TVM_SUPPORT_RANDOM_ENGINE_H_
+#define TVM_SUPPORT_RANDOM_ENGINE_H_
+
+#include <tvm/runtime/logging.h>
+
+#include <cstdint>  // for int64_t
+
+namespace tvm {
+namespace support {
+
+/*!
+ * \brief This linear congruential engine is a drop-in replacement for and strictly corresponds to
+ * std::minstd_rand but designed to be serializable and strictly reproducible. Specifically
+ * implemented for meta schedule but also reusable for other purposes.
+ * \note Part of std::linear_congruential_engine's member functions are not included, for full
+ * member functions of std::minstd_rand, please check out the following link:
```

Comment:
```suggestion
 * \note Parts of std::linear_congruential_engine's member functions are not included. For full
 * member functions of std::minstd_rand, please check out the following link:
```
Should we briefly explain the missing functions and will they be supported in the future?

Reply:
They are not useful so I asked Xiyou to delete them :-)

Reply:
So basically we can say the class is self-contained in terms of the concept `std::uniform_random_bit_generator`

Reply:
Thanks for the noticing difference. We don't intend to support other functions in the future because according to [RandomNumberDistribution](https://en.cppreference.com/w/cpp/named_req/RandomNumberDistribution), the requirement for being used as STL distributions' generator is to be a `std::uniform_random_bit_generator`, which our implementation already is. This note is to show that our class is not a complete implementation of `std::linear_congruential_engine`, i.e., we actually implemented a `std::uniform_random_bit_generator` using a linear congruential engine.